src/client/components/Homepage/SignupForm.jsx

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { createMerchant } from '../../actions/SignUpActions';
+
+class SignupForm extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      businessName: '',
+      email: '',
+      phone: '',
+      username: '',
+      password: '',
+      isVerified: false,
+      admin: false,
+      productCategories: ['General Merchandise'],
+      address: '',
+      hasError: false,
+    }
+
+    this.handleChange = this.handleChange.bind(this);
+    this.createMerchant = this.createMerchant.bind(this);
+  }
+
+  handleChange(e) {
+    /**
+     * clear error if all required fields are filled
+     */
+    if (
+      this.state.businessName !== '' &&
+      this.state.email !== '' &&
+      this.state.username !== '' &&
+      this.state.password !== ''
```

Comment:
### Nitpick
This seems like a long list of conditionals, do you think it can be done better?
Just my thoughts tho, if not we can leave it.

Reply:
Thanks for pointing that out. From the user model, those 4 fields are required and I am making that validation so that the server will not throw the errors back to the front-end. If there's a better way please tell me.